### PR TITLE
Code cleanup for io.micronaut.http.uri

### PR DESCRIPTION
**Pull request by altro3**

### CODE REVIEW

File: http/src/main/java/io/micronaut/http/uri/DefaultUriBuilder.java [hunk 1]
```diff
@@ -44,7 +43,7 @@ class DefaultUriBuilder implements UriBuilder {
     private static final String STRING_PATTERN_SCHEME = "([^:/?#]+):";
     private static final String STRING_PATTERN_USER_INFO = "([^@\\[/?#]*)";
     private static final String STRING_PATTERN_HOST_IPV4 = "[^\\[{/?#:]*";
-    private static final String STRING_PATTERN_HOST_IPV6 = "\\[[\\p{XDigit}\\:\\.]*[%\\p{Alnum}]*\\]";
+    private static final String STRING_PATTERN_HOST_IPV6 = "\\[[\\p{XDigit}:.]*[%\\p{Alnum}]*]";
```
Comment (graemerocher) [inline http/src/main/java/io/micronaut/http/uri/DefaultUriBuilder.java:49]: can you explain the changes to these patterns?

Comment (altro3) [reply http/src/main/java/io/micronaut/http/uri/DefaultUriBuilder.java:49]: Just redundant characters escape. This is just a simplification of the expression, removed unnecessary escapes

**Comment by altro3** [comment]
Also added minor optimization for decoding parameters from URI: `QueryStringDecoder.decodeParams` - without creating `QueryStringDecoder` instance

File: http/src/main/java/io/micronaut/http/uri/QueryStringDecoder.java [hunk 2]
```diff
@@ -123,7 +123,7 @@ final class QueryStringDecoder {
     QueryStringDecoder(String uri, Charset charset, boolean hasPath, int maxParams) {
         this.uri = Objects.requireNonNull(uri, "uri");
         this.charset = Objects.requireNonNull(charset, "charset");
-        this.maxParams = Objects.requireNonNull(maxParams, "maxParams");
+        this.maxParams = maxParams;
```
Comment (altro3) [inline http/src/main/java/io/micronaut/http/uri/QueryStringDecoder.java:126]: don't need to check primitive type to non null value

File: http/src/main/java/io/micronaut/http/uri/DefaultUriBuilder.java [hunk 3]
```diff
@@ -298,15 +297,15 @@ public String toString() {
...
+        var builder = new StringBuilder();
         String scheme = this.scheme;
         String host = this.host;
         if (StringUtils.isNotEmpty(scheme)) {
             if (isTemplate(scheme, values)) {
                 scheme = UriTemplate.of(scheme).expand(values);
             }
             builder.append(scheme)
-                   .append(":");
+                   .append(':');
```
Comment (altro3) [inline http/src/main/java/io/micronaut/http/uri/DefaultUriBuilder.java:311]: Use chars better than Strings

File: http/src/main/java/io/micronaut/http/uri/DefaultUriBuilder.java [hunk 4]
```diff
@@ -148,7 +147,7 @@ class DefaultUriBuilder implements UriBuilder {
 
                 this.path = new StringBuilder(path);
                 if (query != null) {
-                    final Map parameters = new QueryStringDecoder(uri.toString()).parameters();
+                    final Map parameters = QueryStringDecoder.decodeParams(uri.toString());
```
Comment (altro3) [inline http/src/main/java/io/micronaut/http/uri/DefaultUriBuilder.java:153]: Added new method to decode params without creating QueryStringDecoder instance

File: http/src/main/java/io/micronaut/http/uri/DefaultUriBuilder.java [hunk 5]
```diff
@@ -120,7 +119,7 @@ class DefaultUriBuilder implements UriBuilder {
                     this.host = host;
                 }
                 if (port != null) {
-                    this.port = Integer.valueOf(port);
+                    this.port = Integer.parseInt(port);
```
Comment (altro3) [inline http/src/main/java/io/micronaut/http/uri/DefaultUriBuilder.java]: don't need to boxing/ unboxing

File: http/src/main/java/io/micronaut/http/uri/DefaultUriBuilder.java [hunk 6]
```diff
@@ -44,16 +43,16 @@ class DefaultUriBuilder implements UriBuilder {
...
-    private static final String STRING_PATTERN_PORT = "(\\d*(?:\\{[^/]+?\\})?)";
+    private static final String STRING_PATTERN_PORT = "(\\d*(?:\\{[^/]+?})?)";
     private static final String STRING_PATTERN_PATH = "([^#?]*)";
     private static final String STRING_PATTERN_QUERY = "([^#]*)";
     private static final String STRING_PATTERN_REMAINING = "(.*)";
 
     // Regex patterns that matches URIs. See RFC 3986, appendix B
     private static final Pattern PATTERN_SCHEME = Pattern.compile("^" + STRING_PATTERN_SCHEME + "//.*");
-    private static final Pattern PATTERN_FULL_PATH = Pattern.compile("^([^#\\?]*)(\\?([^#]*))?(\\#(.*))?$");
+    private static final Pattern PATTERN_FULL_PATH = Pattern.compile("^([^#?]*)(\\?([^#]*))?(#(.*))?$");
```
Comment (altro3) [inline http/src/main/java/io/micronaut/http/uri/DefaultUriBuilder.java:58]: Just remove redundant escpape characters

**Comment by altro3** [comment]
Added some clarification to my changes

**Review by dstepanov — APPROVED** [review]

**Comment by altro3** [comment]
@graemerocher ping

**Review by graemerocher — APPROVED** [review]

**Comment by graemerocher** [comment]
Thanks for the contribution!

**Pull request merged**